From c8c543febf29491562e32a6c21805ca2af963c2f Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 20 Jan 2025 18:34:47 +0800
Subject: [PATCH 3/8] loonggpu: pass string literal parameter to
 MODULE_IMPORT_NS for Linux >= 6.13

With commit cdd30ebb1b9f ("module: Convert symbol namespace to string
literal"), introduced since Linux 6.13-rc2, the macro `MODULE_IMPORT_NS'
uses string literal as parameter.

Convert parameter `DMA_BUF' to string literal for Linux >= 6.13.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 gsgpu/gsgpu_display.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gsgpu/gsgpu_display.c b/gsgpu/gsgpu_display.c
index 90bc2c4..8bcc5a2 100644
--- a/gsgpu/gsgpu_display.c
+++ b/gsgpu/gsgpu_display.c
@@ -14,9 +14,14 @@
 #include <linux/dma-buf.h>
 #include "gsgpu_helper.h"
 #include "gsgpu_display.h"
+#include <linux/module.h>
 #if defined (MODULE_IMPORT_NS)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 13, 0))
+MODULE_IMPORT_NS("DMA_BUF");
+#else
 MODULE_IMPORT_NS(DMA_BUF);
 #endif
+#endif
 
 static void gsgpu_display_flip_callback(struct dma_fence *f,
 					 struct dma_fence_cb *cb)
-- 
2.48.1

